### PR TITLE
sql: remove calls to errors.New in parser

### PR DESCRIPTION
**Pull request by justinj**

### CODE REVIEW

File: pkg/sql/parser/col_types.go [hunk 1]
```diff
@@ -207,7 +206,7 @@ func LimitDecimalWidth(d *apd.Decimal, precision, scale int) error {
 	}
 	// Use +1 here because it is inverted later.
 	if scale < math.MinInt32+1 || scale > math.MaxInt32 {
-		return errors.New("scale out of range")
+		return pgerror.NewError(pgerror.CodeNumericValueOutOfRangeError, "scale out of range")
```
Comment (knz) [inline pkg/sql/parser/col_types.go]: Pull this into a global var above.

File: pkg/sql/parser/pg_builtins.go [hunk 2]
```diff
@@ -86,8 +85,10 @@ func makeTypeIOBuiltin(argTypes typeList, returnType Type) []Builtin {
 		{
 			Types:      argTypes,
 			ReturnType: fixedReturnType(returnType),
-			fn:         func(_ *EvalContext, _ Datums) (Datum, error) { return nil, errors.New("unimplemented") },
-			Info:       notUsableInfo,
+			fn: func(_ *EvalContext, _ Datums) (Datum, error) {
+				return nil, pgerror.NewError(pgerror.CodeFeatureNotSupportedError, "unimplemented")
```
Comment (knz) [inline pkg/sql/parser/pg_builtins.go]: ditto

File: pkg/sql/parser/txn.go [hunk 3]
```diff
@@ -120,19 +119,19 @@ func (node *TransactionModes) Format(buf *bytes.Buffer, f FmtFlags) {
 func (node *TransactionModes) merge(other TransactionModes) error {
 	if other.Isolation != UnspecifiedIsolation {
 		if node.Isolation != UnspecifiedIsolation {
-			return errors.New("isolation level specified multiple times")
+			return pgerror.NewError(pgerror.CodeSyntaxError, "isolation level specified multiple times")
```
Comment (knz) [inline pkg/sql/parser/txn.go]: ditto, here and below

File: pkg/sql/parser/type_check.go [hunk 4]
```diff
@@ -466,7 +466,7 @@ func (expr *FuncExpr) TypeCheck(ctx *SemaContext, desired Type) (TypedExpr, erro
 		}
 		for i, orderBy := range expr.WindowDef.OrderBy {
 			if orderBy.OrderType != OrderByColumn {
-				return nil, errors.New("ORDER BY INDEX in window definition is not supported")
+				return nil, pgerror.NewError(pgerror.CodeFeatureNotSupportedError, "ORDER BY INDEX in window definition is not supported")
```
Comment (knz) [inline pkg/sql/parser/type_check.go]: ditto, here and below

**Review by knz — COMMENTED** [review]
Nice!!

**Comment by jordanlewis** [comment]
LGTM thanks for doing that!

**Comment by justinj** [comment]
Review status: 0 of 10 files reviewed at latest revision, 4 unresolved discussions.

---

*[pkg/sql/parser/col_types.go, line 209 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18742#-KutikRztWbAj7kAprrM-r1-209:-KuuHrA3z2ln-nVaHRYH:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/97785e2f1d2b3d6da1cb9880cb72814c4f8e64e8/pkg/sql/parser/col_types.go#L209)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Pull this into a global var above.
</blockquote></details>

Done.

---

*[pkg/sql/parser/pg_builtins.go, line 89 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18742#-KutikS-74ceKe1oUpg0-r1-89:-KuuI1b0Td46EbgIEYxR:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/97785e2f1d2b3d6da1cb9880cb72814c4f8e64e8/pkg/sql/parser/pg_builtins.go#L89)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

---

*[pkg/sql/parser/txn.go, line 122 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18742#-KutikS-74ceKe1oUpg2-r1-122:-KuuIQt9ziiR2Kta0Ck9:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/97785e2f1d2b3d6da1cb9880cb72814c4f8e64e8/pkg/sql/parser/txn.go#L122)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

ditto, here and below
</blockquote></details>

Done.

---

*[pkg/sql/parser/type_check.go, line 469 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/18742#-KutikS-74ceKe1oUpg3-r1-469:-KuuIjy8w677ybn8TGBp:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/97785e2f1d2b3d6da1cb9880cb72814c4f8e64e8/pkg/sql/parser/type_check.go#L469)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

ditto, here and below
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18742)*

**Comment by knz** [comment]
:lgtm:

---

Reviewed 6 of 10 files at r1, 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/18742#-:-KuuWv-_zBy8vcSsl1Sd:bnfp4nl)*

**Comment by justinj** [comment]
TFTRs!

**Pull request merged**